curation/imports/scoring_file.py

```diff
@@ -155,3 +155,92 @@ def update_scoring_file(self):
+        elif genome_build in ('GRCh38', 'hg38'):
+            build_version = '38'
+        elif genome_build == 'NR':
+            report_func('Genome build = NR, variant positions validation ignored.')
+            return failed_qc
+        else:
+            report_func(f'Genome build = "{genome_build}", not detected as 37 or 38, variant positions validation ignored.')
+            return failed_qc
+        report_func(f'Build version = {build_version}')
+
+        new_score_file = f'{self.new_score_file_path}/{score.id}.txt.gz'
+
+        # Reading the scoring file
+        df = pd.read_csv(new_score_file,
+                         sep="\t",
+                         comment='#',
+                         usecols=lambda c: c in usecols,
+                         dtype={"rsID": 'string', "chr_name": 'string', "chr_position": 'Int64',
+                                "effect_allele": 'string', "other_allele": 'string'})
+
+        if 'chr_position' not in df.columns:
+            report_func('No chr_position column. Variant position QC will be skipped')
+            return failed_qc
+
+        # Headers for alleles. other_allele is optional, but should be tested if exists as we don't know which allele is the reference one.
+        alleles_headers = ['effect_allele']
+        if 'other_allele' in df.columns:
+            alleles_headers.append('other_allele')
+
+        n_requests = self.variant_positions_qc_config['n_requests']
+
+        if 'rsID' in df.columns:
+            max_request_size = self.variant_positions_qc_config['ensembl_max_variation_req_size']
+            map_variants_func = map_rsids
+        else:
+            if len(alleles_headers) == 1:
+                # 1 allele column is not enough for sequence-based validation, as we don't know if the allele is ref or alt.
+                report_func('Only 1 allele column with no rsID. Variant position QC will be skipped')
+                return failed_qc
+            max_request_size = self.variant_positions_qc_config['ensembl_max_sequence_req_size']
+            map_variants_func = map_variants_to_reference_genome
+
+        errors = []
+        variant_slices = sample_df(df, n_requests, max_request_size, alleles_headers, False, errors)
+
+        match = 0
+        mismatch = 0
+        for variants in variant_slices:
+            results = map_variants_func(variants, build_version)
+            match += results['match']
+            mismatch += results['mismatch']
+            errors = errors + results['errors']
+
+        final_report = '''Results:
+         - Matches: {0}
+         - Mismatches: {1}'''
+        report_func(final_report.format(str(match), str(mismatch)))
+
+        # Fail if low match rate
+        if match+mismatch == 0:
+            report_func('WARNING: No match data!')
+        else:
+            match_rate: float = float(match) / (match + mismatch)
+            report_func(f'Match rate: {match_rate}')
+            if match_rate < 0.9:
```

Comment:
Should we store the threshold (i.e. `0.9`) in a variable, preferably in the config file ?